WordPress/Classes/Services/AccountService.m

```diff
@@ -287,4 +287,24 @@ - (void)updateAccount:(WPAccount *)account withUserDetails:(RemoteUser *)userDet
     [[ContextManager sharedInstance] saveContext:self.managedObjectContext];
 }
 
+- (void)purgeSelfHostedAccount:(WPAccount *)account
+{
+    if (!account.isWpcom && [account.blogs count] == 0) {
+        [self.managedObjectContext deleteObject:account];
+    }
+}
+
+- (void)purgeJetpackAccount:(WPAccount *)account
+{
+    WPAccount *defaultAccount = [self defaultWordPressComAccount];
+    if (account
+        && [account.jetpackBlogs count] == 0
+        && [account.visibleBlogs count] == 0
+        && ![defaultAccount isEqual:account]) {
+        DDLogWarn(@"Removing jetpack account %@ since the last blog using it is being removed", account.username);
```

Comment:
"...since the last blog using it is being removed" - this method doesn't know if this statement is true.

Although since this is just a log message it's probably not very important and we may leave it like that.  Your call.